soa-derive-internal/src/refs.rs

```diff
@@ -16,23 +16,79 @@ pub fn derive(input: &Input) -> TokenStream {
     let ref_doc_url = format!("[`{0}`](struct.{0}.html)", ref_name);
     let ref_mut_doc_url = format!("[`{0}`](struct.{0}.html)", ref_mut_name);
 
-    let fields_names = input.fields.iter()
-                                   .map(|field| field.ident.clone().unwrap())
-                                   .collect::<Vec<_>>();
-    let fields_names_1 = &fields_names;
-    let fields_names_2 = &fields_names;
+    let get_ref_field_doc = |field_ident: &Ident| {
```

Comment:
Is this function (and get_ref_mut_field_doc) only use once below? Why not write them directly inside the call to `fields_seq`?